jwst/photom/photom.py

```diff
@@ -468,7 +470,32 @@ def calc_miri(self, ftab):
                 row = find_row(ftab.phot_table, fields_to_match)
                 if row is None:
                     return
-            self.photom_io(ftab.phot_table[row])
+                self.photom_io(ftab.phot_table[row])
+                
+            # Check if reference file contains the coefficients for the time-dependent correction of the PHOTOM value
+            try:
+                ftab.getarray_noinit("timecoeff")
+                log.info("Applying the time-dependent correction to the PHOTOM value.")
+
+                mid_time = self.input.meta.exposure.mid_time
+                photom_corr = miri_imager.time_corr_photom(ftab.timecoeff[row], mid_time)
+
+                data = np.array(
+                    [(self.filter, self.subarray, ftab.phot_table[row]['photmjsr']+photom_corr, ftab.phot_table[row]['uncertainty'])],
+                    dtype=[
+                        ("filter", "O"),
+                        ("subarray", "O"),
+                        ("photmjsr", "<f4"),
+                        ("uncertainty", "<f4")
+                        ],
+                )
+                fftab = MirImgPhotomModel(phot_table=data)
```

Comment:
Suggest changing this to `fftab = datamodels.MirImgPhotomModel(phot_table=data)` and then don't import `MirImgPhotomModel` at the beginning of the module.